urbs.py

```diff
@@ -964,7 +973,17 @@ def res_process_capacity_rule(m, sit, pro):
     return (m.process.loc[sit, pro]['cap-lo'],
             m.cap_pro[sit, pro],
             m.process.loc[sit, pro]['cap-up'])
-
+            
+# used roof area <= roof area
+def res_process_roof_area_rule(m, sit, pro):
+    for p in m.pro_tuples:
+        try:
+            float(m.process.loc[p]['area-per-cap']) 
+            return (0,
```

Comment:
Instead of the tuple form, I would recommend using the `sum(areas) <= m.sit.loc['area_roof']` form. I don't see any necessity to explicitly forbid negative area sums. Some technology could even reduce area consumption, maybe by allowing stacking.

Naming concern: `area_roof` as a site attribute is unnecessarily carrying semantics: a site has an `area`. `area_roof` or other `_suffix` types should either become a full set of their own (leading to a MultiIndex in spreadsheet `Site`) or (as I recommend) be omitted for now. I don't see an immediate need to specify exactly two different area values per site.

More general thought: if more capacity-like coupling restrictions among processes might be required, one should first dedicate more whiteboard time to properly model the required features, instead of adding attributes. (Adding attributes is always easy, but is to be paid 'in interest' later by model users.)